storage/client/src/test/java/com/azure/storage/blob/ContainerAPITest.groovy

```diff
@@ -1031,6 +1031,21 @@ class ContainerAPITest extends APISpec {
         response.iterator().size() == 4
     }*/
 
+    def "List blobs flat simple"() {
+        setup:
+        // Create 10 page blobs in the container
+        for (int i = 0; i < 10; i++) {
+            PageBlobClient bu = cu.getPageBlobClient(generateBlobName())
+            bu.create(512)
+        }
+
+        Iterable<BlobItem> blobs = cu.listBlobsFlat()
```

Comment:
Can you pass a ListBlobsOptions here that sets maxResults to like 3 so we can verify the autopagination? And then instead of checking blobs.size(), update a counter while you iterate over the items (iteratable.size() is a groovy thing that won't actually work here because of how the list is broken up over several rest calls).

Reply:
thank you good point, I tried with ListBlobsOptions that limits result per page to 3. 

Regarding iteratable.size() - I was debugging this, it seems iterable is created from Flux which asynchronously pull next pages. The iterable.size() causing the Flux to drain which result in making multiple calls. Please let me know we still want to go with counter based approach.

Reply:
That makes sense. Groovy continues to surprise me :) I'm fine with using size if it does the right thing.

Reply:
I'm fine for you to go ahead and merge